src/main/java/org/folio/s3/client/FolioS3Client.java

```diff
@@ -101,4 +104,72 @@ public interface FolioS3Client {
    * Creates bucket. Bucket name should be declared in {@link S3ClientProperties}
    */
   void createBucketIfNotExists();
+
+  /**
+   * Initiates a multipart upload, returning the upload ID.
+   *
+   * @param path - the path to the file on S3-compatible storage
+   * @return the multipart upload ID
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateMultipartUpload.html
+   */
+  String initiateMultipartUpload(String path);
+
+  /**
+   * Gets a presigned URL to PUT a part of a multipart upload
+   *
+   * @param path       - the path to the file on S3-compatible storage
+   * @param uploadId   - the upload ID from
+   *                   {@link #initiateMultipartUpload(String)}
+   * @param partNumber - the part number of the part to upload, starts at 1
+   * @return the presigned URL
+   * @see {@link #initiateMultipartUpload(String)}
+   */
+  String getPresignedMultipartUploadUrl(
+      String path,
+      String uploadId,
+      int partNumber);
+
+  /**
+   * Uploads a part of a multipart upload from a local file
+   *
+   * @param path       - the path to the file on S3-compatible storage
+   * @param uploadId   - the upload ID from
+   *                   {@link #initiateMultipartUpload(String)}
+   * @param partNumber - the part number of the part to upload, starts at 1
+   * @param filename   - the local uploaded file on disk
+   * @return the upload's eTag
+   * @see {@link #initiateMultipartUpload(String)}
+   */
+  String uploadMultipartPart(
```

Comment:
just one idea: maybe the name of the method "uploadMultipart" is enough, because this method has partNumber as a parameter? thanks

Reply:
It is a bit redundant, but since there are so many options when dealing with multipart uploads (create, upload part, copy part, complete multipart, abort multipart, list parts, list multiparts, ...), so in my opinion it's best to have the slightly clunkier but more specific method name.

Reply:
ok, thanks